lib/pagseguro/transaction/serializer.rb

```diff
@@ -34,9 +34,9 @@ def serialize_status_history
       def serialize_general(data)
         data[:code] = xml.at_css("code").text
         data[:reference] = xml.css("reference").text
-        data[:type_id] = xml.at_css("type").text
         data[:payment_link] = xml.css("paymentLink").text
-        data[:status] = xml.at_css("status").text
+        data[:type_id] = xml.at_css("type").text if xml.at_css('type')
+        data[:status] = xml.at_css("status").text if xml.at_css('status')
```

Comment:
Yes, but I think this won't solve the problem, because the ```at_css``` will return the first result, and the first result can be two types of status different.

Look at the problem:

```
<xmlTag>
  <status>value</status>
  <anotherXmlTag>
    <status>anotherValue</status>
  </anotherXmlTag>
</xmlTag>
```

The ```at_css``` just try to get the first result, no matter what.